app/utilities/PersistenceManager.java

```diff
@@ -87,6 +88,23 @@ public static Token getToken(String token){
     }
 
     /**
+     * Persists restaurant into database.
+     *
+     * @param restaurant the Restaurant to be persisted
+     */
+    public static void saveRestaurant(Restaurant restaurant){
+        restaurant.getAddress().setRestaurantId(Restaurant.find.all().size() + 1);
```

Comment:
This is a bad way of setting ID. ID should be generated via sequence on db side, not like this. This is not the practice and it doesn't support concurrency.    

@GeneratedValue(strategy = GenerationType.AUTO) is supported with auto_increment on db side. You should not worry about the ID! Same for all models.